src/main/java/io/supertokens/userroles/UserRoles.java

```diff
@@ -40,6 +41,44 @@ public static void addRoleToUser(Main main, String userId, String role)
         }
     }
 
+    // create a new role if it doesn't exist and add permissions to the role
+    public static void setRole(Main main, String role, String[] permissions)
+            throws StorageQueryException, StorageTransactionLogicException, UnknownRoleException {
+        UserRolesSQLStorage storage = StorageLayer.getUserRolesStorage(main);
+        try {
+            storage.startTransaction(con -> {
+                try {
+                    storage.createNewRole_Transaction(con, role);
+                } catch (DuplicateRoleException e) {
+                    // ignore exception
+                }
+                if (permissions != null) {
+                    for (int i = 0; i < permissions.length; i++) {
+                        try {
+                            storage.addPermissionToRole_Transaction(con, role, permissions[i]);
+                        } catch (DuplicateRolePermissionMappingException e) {
+                            // ignore exception
+                        } catch (UnknownRoleException e) {
+                            throw new StorageTransactionLogicException(e);
```

Comment:
In this case, we ignore the error since it should never actually come here. And we write a comment explaining why we are ignoring it